kuma/demos/helpers.py

```diff
@@ -162,30 +167,20 @@ def search_form(context):
     return new_context(**locals())
 
 
-bitly_api = None
-
-
-def _get_bitly_api():
-    """Get an instance of the bit.ly API class"""
-    global bitly_api
-    if bitly_api is None:
-        import bitly
-        login = getattr(settings, 'BITLY_USERNAME', '')
-        apikey = getattr(settings, 'BITLY_API_KEY', '')
-        bitly_api = bitly.Api(login, apikey)
-    return bitly_api
-
-
 @register.filter
 def bitly_shorten(url):
     """Attempt to shorten a given URL through bit.ly / mzl.la"""
-    try:
-        # TODO:caching
-        return _get_bitly_api().shorten(url)
-    except:
-        # Just in case the bit.ly service fails or the API key isn't
-        # configured, fall back to using the original URL.
-        return url
+    cache_key = 'bitly:%s' % url
```

Comment:
iirc the max key size is 250 characters? URLs can be long. Should we md5 that so it's unique but no longer than 32 characters plus the prefix?